### PR TITLE
Fix `no-unexpected-multiline` false positive

### LINKED ISSUE #5148: `no-unexpected-multiline` false positive (eslint 2.0.0)
Using the latest eslint (2.0.0-rc.0).

Rules:

```json
{
  "rules": {
    "no-unexpected-multiline": 2
  }
}
```

Code:

```js
(
  function () {}
)[1]
```

Results in the following false positive:

```
  3:2  error  Unexpected newline between object and [ of property access  no-unexpected-multiline
```

This does not trigger an error in eslint v1.10.3.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30520351-no-unexpected-multiline-false-positive-eslint-2-0-0?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by feross**
Fixes #5148.

### CODE REVIEW

**Comment by feross** [comment]
Tests pass :+1:

File: lib/rules/no-unexpected-multiline.js [hunk 1]
```diff
@@ -24,11 +24,13 @@ module.exports = function(context) {
      */
     function checkForBreakAfter(node, msg) {
         var paren = context.getTokenAfter(node);
+        var beforeParen = node;
         while (paren.value === ")") {
+            beforeParen = paren;
             paren = context.getTokenAfter(paren);
```
Comment (glenjamin) [inline lib/rules/no-unexpected-multiline.js]: I think this variable being called `paren` is confusing, if I'm reading it right it ends up being the next node *after* the paren?

**Comment by ilyavolodin** [comment]
LGTM. Thanks for PR

File: lib/rules/no-unexpected-multiline.js (hunk 1 above)
Comment (glenjamin) [inline lib/rules/no-unexpected-multiline.js]: If i've followed this right, when the `while` is exited
```js
beforeParen = node(")")
paren = node(whatever came next)
```

If this is right, the names should probably be swapped to `paren` and `afterParen`

**Comment by glenjamin** [comment]
Have sent a PR to the PR bikeshedding some variable names to hopefully improve clarity: https://github.com/feross/eslint/pull/1

File: lib/rules/no-unexpected-multiline.js (hunk 1 above)
Comment (glenjamin) [inline lib/rules/no-unexpected-multiline.js]: For future reference, I was not right. The `paren` variable and the comparison to `)` confused me.

**Comment by feross** [comment]
@glenjamin Thanks, merged your PR that clarifies the naming.

**Comment by platinumazure** [comment]
@feross You might need to squash the commits.

**Comment by feross** [comment]
Squashed and updated.

**Comment by gyandeeps** [comment]
LGTM. Thanks

**Pull request merged**